test/gtest/graphapi_operation_rng.cpp

```diff
@@ -0,0 +1,335 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/graphapi/rng.hpp>
+#include <miopen/graphapi/tensor.hpp>
+
+#include <array>
+#include <tuple>
+#include <variant>
+
+#include <gtest/gtest.h>
+
+#include "graphapi_gtest_common.hpp"
+
+namespace miopen {
+
+namespace graphapi {
+
+std::ostream& operator<<(std::ostream& os, std::variant<int64_t, Tensor*> seed)
+{
+    if(seed.index() == 0)
+        return os << std::get<0>(seed);
+    else
+        return os << std::get<1>(seed);
+};
+
+} // namespace graphapi
+
+} // namespace miopen
+
+namespace {
+
+using miopen::graphapi::OperationRng;
+using miopen::graphapi::OperationRngBuilder;
+using miopen::graphapi::Rng;
+using miopen::graphapi::Tensor;
+
+using miopen::graphapi::ValidatedValue;
+
+using DescriptorTuple = std::tuple<bool,
+                                   ValidatedValue<Rng*>,
+                                   ValidatedValue<Tensor*>,
+                                   ValidatedValue<std::variant<int64_t, Tensor*>>,
+                                   ValidatedValue<Tensor*>>;
+
+} // namespace
+
+class GraphApiOperationRngBuilder : public testing::TestWithParam<DescriptorTuple>
+{
+protected:
+    bool attrsValid;
```

Comment:
In class  OperationRng you use prefix "m" for variables. Why in this class you dont use this prefix ?

Reply:
Should we put public section first, and protected, private after publicr ?
Usually, people use the class and need public interface first of all, so its more convinient to put it first.

But this is a topic for discussion of course. Not sure do we have some rules about it/

Reply:
I don't use `m` prefix in GTest Fixtures to emphasize input parameters of a test.
Also I didn't use `m` for `execute` in another Fixture to show that it is a functional object.

Reply:
There will be no public section in this class. It is a Fixture; TEST_P seems to create a descendant class internally and puts its (TEST_P's) body in one of class's methods so protected members will be available, and they are perceived by you as local variables when you write TEST_P body.

Reply:
I thought we want to use "m" prefix for class members, does not matter what this param is intended for. Just to distinguish them with local function variables.
But ok.

Reply:
This is something we haven't agreed as a team on, and we are not enforcing, so you can delay making this change if you want.

Reply:
done.
And in the next PR too.
But I'm not sure we should apply this rule to public members.